### PR TITLE
add not implemented errors where needed

### PR DESCRIPTION
**Pull request by billiford**
- add errors when trying to scale or perform a rolling restart on an unsupported kind
- add error when trying to patch using an unsupported merge type
- add error when trying to resolve unsupported artifact type

### CODE REVIEW

File: pkg/http/core/kubernetes/restart_test.go [hunk 1]
```diff
@@ -73,6 +73,17 @@ var _ = Describe("Restart", func() {
 		})
 	})
 
+	When("the kind is not supported to restarted", func() {
+		BeforeEach(func() {
+			actionConfig.Operation.RollingRestartManifest.ManifestName = "not-supported-kind test-name"
```
Comment (iteaguy) [inline pkg/http/core/kubernetes/restart_test.go:78]: Shuldnt this be?:
actionConfig.Operation.RollingRestartManifest.ManifestName = “not-supported-kind”?

to match the message in Line #83

File: pkg/http/core/kubernetes/scale_test.go [hunk 2]
```diff
@@ -84,6 +84,17 @@ var _ = Describe("Scale", func() {
 		})
 	})
 
+	When("the kind is not supported to scale", func() {
+		BeforeEach(func() {
+			actionConfig.Operation.ScaleManifest.ManifestName = "not-supported-kind test-name"
```
Comment (iteaguy) [inline pkg/http/core/kubernetes/scale_test.go:89]: Shuldnt this be?:
actionConfig.Operation.ScaleManifest.ManifestName = “not-supported-kind”?

to match the message in Line #94

File: pkg/http/core/kubernetes/restart_test.go (hunk 1 above)
Comment (billiford) [reply pkg/http/core/kubernetes/restart_test.go:78]: This is one aspect of the Spinnaker API that I've always found weird. When referencing a manifest by name it is almost always referenced as `<kind> <name>`, which is what I'm doing here.

**Review by iteaguy — CHANGES_REQUESTED** [review]
Added some questions.

**Pull request merged**